### ISSUE TITLE
Error when applying global event rules without explicit dependencies

### ISSUE THREAD

**Issue by cjdaniel** [PagerDuty/terraform-provider-pagerduty#175]
When creating, modifying, or deleting event rules with no explicit dependencies on each other, I get a "409 Conflict" error when applying *some* of the rules. The number of errors decreases with subsequent "apply" or "destroy" commands, eventually resulting in a successful apply.

- If there is only one event rule in the set, this behavior doesn't occur.

- If there are explicit dependencies using `depends_on`, this behavior doesn't occur. (The order of our event rules doesn't matter, since our conditions are very specific; so we don't have any dependencies declared.)

I believe this error may occur due to the way Terraform applies resources concurrently, when it can. In debug output, I can see the 409 error is sent in response to POST requests that are sent in the same millisecond. It doesn't always happen that way, though. I think `depends_on` forces Terraform to apply each rule in sequence rather than trying any at once, avoiding this behavior.

The PagerDuty Global Event Rules API documentation states there is a global `object_version` that changes each time the ruleset is updated. Maybe if this changes at the same time another change is being processed, the error occurs. Maybe this is a bug for the PagerDuty API -- it isn't thread-safe? I believe this type of problem has been discussed in https://github.com/hashicorp/terraform/issues/14550

If it makes sense, I think this provider should work around the issue by ensuring event rules are applied individually and never in parallel.

### Terraform Version
Terraform v0.12.18

### Affected Resource(s)
- pagerduty_event_rule

### Terraform Configuration Files
```hcl
resource "pagerduty_event_rule" "bug1" {
  action_json = jsonencode([
    [
      "annotate",
      "This is a test 1",
    ],
  ])

  condition_json = jsonencode([
    "and",
    [
      "matches",
      [
        "path",
        "service",
      ],
      "^foo-1",
    ],
    [
      "equals",
      [
        "path",
        "client",
      ],
      "bar",
    ],
  ])
}

resource "pagerduty_event_rule" "bug2" {
  action_json = jsonencode([
    [
      "annotate",
      "This is a test 2",
    ],
  ])

  condition_json = jsonencode([
    "and",
    [
      "matches",
      [
        "path",
        "service",
      ],
      "^foo-2",
    ],
    [
      "equals",
      [
        "path",
        "client",
      ],
      "bar",
    ],
  ])
}
```

### Debug Output
`TF_LOG=TRACE terraform apply` output:
https://gist.github.com/cjdaniel/f492f26203ee9f860cc1ba256dade322

### Expected Behavior
The rules should be created with an undetermined order.

### Actual Behavior
An error was given by the PagerDuty API during apply:
```
Error: POST API call to https://api.pagerduty.com/event_rules failed: 409 Conflict

  on rules.tf line 1, in resource "pagerduty_event_rule" "bug1":
   1: resource "pagerduty_event_rule" "bug1" {
```
Subsequent applies are required to get to a clean state.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Write a configuration including more than one event rule, with no explicit dependencies.
2. `terraform apply`
3. If the 409 error doesn't occur, `destroy` and try again. It sometimes won't occur.

### References
https://github.com/hashicorp/terraform/issues/14550

**Comment by ffran09** [comment]
We also face the same issue. Does anyone know how we can get this bug prioritized?

**Comment by stmcallister** [comment]
Completely agree that there needs to be a better way to handle Event Rules. In the short term, I think the provider needs some retry logic to possibly get around the 409 messages. But, I also know that there will be a RuleSets API generally available by April of this year. Once that API is publicly available we'll work on getting the resource built in the provider.

**Comment by pdecat** [comment]
Hi @stmcallister, 

> I also know that there will be a RuleSets API generally available by April of this year. Once that API is publicly available we'll work on getting the resource built in the provider.

We are currently looking into implementing specific Service Event Rules, not Global Event Rules.
I did not find an mention of such event rules in the API reference.
Can you confirm there isn't any available at this time?

Will those also be supported by the RuleSets API you mention?

Should I open a separate issue to track this?

**Comment by matt-canty-dragon** [comment]
Thought I'd note that we are being advised by PagerDuty to use global rules over service rules. Until I know better I will be sticking with their advice, however this issue is quite disruptive.

I note that service rules via Terraform will have the same issue.

**Comment by matt-canty-dragon** [comment]
Can someone clarify whether https://www.terraform.io/docs/providers/pagerduty/r/ruleset_rule.html has the same issue?

I have just added a load of rules and it took some 16 attempts to create them all. Turns out edits also have this issue! I thought it would just be on create.

Hoping I can use `ruleset_rule` and hardcode the global ruleset ID.

Thanks

**Comment by DPM123** [comment]
Same issue here on version 0.12.20
Changing parallelism to 1 removes the 409 conflicts


2020-06-11T15:09:08.865+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: -----------------------------------------------------
2020-06-11T15:09:08.865+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: 2020/06/11 15:09:08 [TRACE] Waiting 500ms before next try
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: 2020/06/11 15:09:08 [DEBUG] PagerDuty API Response Details:
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: ---[ RESPONSE ]--------------------------------------
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: HTTP/2.0 409 Conflict
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: Content-Length: 40
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: Cache-Control: max-age=0, private, must-revalidate
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: Content-Type: application/vnd.pagerduty+json; charset=UTF-8
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: Date: Thu, 11 Jun 2020 14:09:08 GMT
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: Server: nginx
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4:
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: {
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: "error": {
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: "message": "Version Conflict"
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: }
2020-06-11T15:09:08.939+0100 [DEBUG] plugin.terraform-provider-pagerduty_v1.7.2_x4: }

**Comment by stmcallister** [comment]
We've recently implemented several updates relating to Global Event Rules and Service Event Rules that should have addressed these issues. Keyword is "should". Is anyone on this thread still experiencing these issues using at least v1.9.4 of the provider?

**Comment by cjdaniel** [comment]
Using 1.9.4, I no longer see this issue with the example code provided in my initial report. I am pretty sure that solves it! Thank you!

**Comment by matt-canty-dragon** [comment]
I have tested `pagerduty_ruleset_rule` and I think that it is working. Here is my console output:

# Added 3 rules
```
➜  test-tf git:(master) ✗ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.pagerduty_ruleset.example: Refreshing state...

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # pagerduty_ruleset_rule.a will be created
  + resource "pagerduty_ruleset_rule" "a" {
      + id      = (known after apply)
      + ruleset = "6c40722c-ffb8-4c0c-b3b3-af035353999f"

      + actions {

          + severity {
              + value = "critical"
            }
        }

      + conditions {
          + operator = "and"

          + subconditions {
              + operator = "equals"

              + parameter {
                  + path  = "vendor"
                  + value = "a"
                }
            }
        }
    }

  # pagerduty_ruleset_rule.b will be created
  + resource "pagerduty_ruleset_rule" "b" {
      + id      = (known after apply)
      + ruleset = "6c40722c-ffb8-4c0c-b3b3-af035353999f"

      + actions {

          + severity {
              + value = "critical"
            }
        }

      + conditions {
          + operator = "and"

          + subconditions {
              + operator = "equals"

              + parameter {
                  + path  = "vendor"
                  + value = "b"
                }
            }
        }
    }

  # pagerduty_ruleset_rule.c will be created
  + resource "pagerduty_ruleset_rule" "c" {
      + id      = (known after apply)
      + ruleset = "6c40722c-ffb8-4c0c-b3b3-af035353999f"

      + actions {

          + severity {
              + value = "critical"
            }
        }

      + conditions {
          + operator = "and"

          + subconditions {
              + operator = "equals"

              + parameter {
                  + path  = "vendor"
                  + value = "c"
                }
            }
        }
    }

Plan: 3 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.

➜  test-tf git:(master) ✗ terraform apply -auto-approve
data.pagerduty_ruleset.example: Refreshing state...
pagerduty_ruleset_rule.c: Creating...
pagerduty_ruleset_rule.b: Creating...
pagerduty_ruleset_rule.a: Creating...
pagerduty_ruleset_rule.b: Creation complete after 1s [id=8e025191-9dca-434b-8680-f3397198ec9f]
pagerduty_ruleset_rule.c: Creation complete after 2s [id=9f897169-1798-4807-86f5-2c6c8ae1c09f]
pagerduty_ruleset_rule.a: Creation complete after 2s [id=fd8d04ff-de0e-4986-bb1c-da7e0d94d9d7]

Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
```

# Removed the "Middle" rule

```
➜  test-tf git:(master) ✗ terraform plan               
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.pagerduty_ruleset.example: Refreshing state...
pagerduty_ruleset_rule.c: Refreshing state... [id=9f897169-1798-4807-86f5-2c6c8ae1c09f]
pagerduty_ruleset_rule.b: Refreshing state... [id=8e025191-9dca-434b-8680-f3397198ec9f]
pagerduty_ruleset_rule.a: Refreshing state... [id=fd8d04ff-de0e-4986-bb1c-da7e0d94d9d7]

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place
  - destroy

Terraform will perform the following actions:

  # pagerduty_ruleset_rule.b will be updated in-place
  ~ resource "pagerduty_ruleset_rule" "b" {
        disabled = false
        id       = "8e025191-9dca-434b-8680-f3397198ec9f"
      - position = 2 -> null
        ruleset  = "6c40722c-ffb8-4c0c-b3b3-af035353999f"

        actions {

            severity {
                value = "critical"
            }
        }

        conditions {
            operator = "and"

            subconditions {
                operator = "equals"

                parameter {
                    path  = "vendor"
                    value = "b"
                }
            }
        }
    }

  # pagerduty_ruleset_rule.c will be destroyed
  - resource "pagerduty_ruleset_rule" "c" {
      - disabled = false -> null
      - id       = "9f897169-1798-4807-86f5-2c6c8ae1c09f" -> null
      - position = 1 -> null
      - ruleset  = "6c40722c-ffb8-4c0c-b3b3-af035353999f" -> null

      - actions {

          - severity {
              - value = "critical" -> null
            }
        }

      - conditions {
          - operator = "and" -> null

          - subconditions {
              - operator = "equals" -> null

              - parameter {
                  - path  = "vendor" -> null
                  - value = "c" -> null
                }
            }
        }
    }

Plan: 0 to add, 1 to change, 1 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.

➜  test-tf git:(master) ✗ terraform apply -auto-approve
data.pagerduty_ruleset.example: Refreshing state...
pagerduty_ruleset_rule.c: Refreshing state... [id=9f897169-1798-4807-86f5-2c6c8ae1c09f]
pagerduty_ruleset_rule.a: Refreshing state... [id=fd8d04ff-de0e-4986-bb1c-da7e0d94d9d7]
pagerduty_ruleset_rule.b: Refreshing state... [id=8e025191-9dca-434b-8680-f3397198ec9f]
pagerduty_ruleset_rule.c: Destroying... [id=9f897169-1798-4807-86f5-2c6c8ae1c09f]
pagerduty_ruleset_rule.b: Modifying... [id=8e025191-9dca-434b-8680-f3397198ec9f]
pagerduty_ruleset_rule.c: Destruction complete after 0s
pagerduty_ruleset_rule.b: Modifications complete after 1s [id=8e025191-9dca-434b-8680-f3397198ec9f]

Apply complete! Resources: 0 added, 1 changed, 1 destroyed.
```

**Comment by matt-canty-dragon** [comment]
2nd test. I created 100 rules using `count`. 100 rules were created and destroyed quickly and without error. 

```
➜  test-tf git:(master) ✗ terraform apply -auto-approve
data.pagerduty_ruleset.example: Refreshing state...
pagerduty_ruleset_rule.a[0]: Refreshing state... [id=fd8d04ff-de0e-4986-bb1c-da7e0d94d9d7]
pagerduty_ruleset_rule.a[1]: Refreshing state... [id=245da58a-d4a7-44e8-986d-a1487f59ca6e]
pagerduty_ruleset_rule.a[55]: Creating...
pagerduty_ruleset_rule.a[47]: Creating...
pagerduty_ruleset_rule.a[18]: Creating...
pagerduty_ruleset_rule.a[82]: Creating...
pagerduty_ruleset_rule.a[24]: Creating...
pagerduty_ruleset_rule.a[38]: Creating...
pagerduty_ruleset_rule.a[93]: Creating...
pagerduty_ruleset_rule.a[75]: Creating...
pagerduty_ruleset_rule.a[57]: Creating...
pagerduty_ruleset_rule.a[1]: Modifying... [id=245da58a-d4a7-44e8-986d-a1487f59ca6e]
pagerduty_ruleset_rule.a[47]: Creation complete after 1s [id=89207966-e67b-4188-8a32-fc5a1f3e5086]
pagerduty_ruleset_rule.a[51]: Creating...
pagerduty_ruleset_rule.a[55]: Creation complete after 1s [id=fcc1ccac-d5d1-4f72-8547-9c8be891252d]
pagerduty_ruleset_rule.a[14]: Creating...
pagerduty_ruleset_rule.a[24]: Creation complete after 2s [id=455c59dd-d82a-494f-bf40-247e64e54e06]
pagerduty_ruleset_rule.a[39]: Creating...
pagerduty_ruleset_rule.a[1]: Modifications complete after 2s [id=245da58a-d4a7-44e8-986d-a1487f59ca6e]
pagerduty_ruleset_rule.a[57]: Creation complete after 2s [id=fd6cead2-177e-4bfc-a818-ed66100bba6a]
pagerduty_ruleset_rule.a[61]: Creating...
pagerduty_ruleset_rule.a[6]: Creating...
pagerduty_ruleset_rule.a[93]: Creation complete after 2s [id=43a53844-4738-44a9-b614-9b71ca420b83]
pagerduty_ruleset_rule.a[68]: Creating...
pagerduty_ruleset_rule.a[51]: Creation complete after 1s [id=7d28eed7-8c4b-4f1e-90ef-33298f70deb3]
pagerduty_ruleset_rule.a[19]: Creating...
pagerduty_ruleset_rule.a[39]: Creation complete after 1s [id=29ec7e33-be89-4f36-8c56-e08b0f709435]
pagerduty_ruleset_rule.a[83]: Creating...
pagerduty_ruleset_rule.a[6]: Creation complete after 1s [id=35d18ee4-57b2-4049-a4bd-709a17920b0c]
pagerduty_ruleset_rule.a[48]: Creating...
pagerduty_ruleset_rule.a[61]: Creation complete after 1s [id=060f410c-02e7-4137-8569-c94d04cd160e]
pagerduty_ruleset_rule.a[76]: Creating...
pagerduty_ruleset_rule.a[75]: Creation complete after 3s [id=ab4eece5-52b4-4f7f-8f49-13bc8f8de907]
pagerduty_ruleset_rule.a[88]: Creating...
pagerduty_ruleset_rule.a[68]: Creation complete after 1s [id=496baada-2df4-4bee-bc30-1c8508665048]
pagerduty_ruleset_rule.a[33]: Creating...
pagerduty_ruleset_rule.a[14]: Creation complete after 3s [id=40f2d89f-8eb3-49d4-a546-81a4f6f4d186]
pagerduty_ruleset_rule.a[23]: Creating...
pagerduty_ruleset_rule.a[76]: Creation complete after 1s [id=38fe3be6-fb7d-4847-ad18-847512cb3bae]
pagerduty_ruleset_rule.a[27]: Creating...
pagerduty_ruleset_rule.a[83]: Creation complete after 1s [id=9b909eef-531d-4b9f-a062-3339688cfc7e]
pagerduty_ruleset_rule.a[10]: Creating...
pagerduty_ruleset_rule.a[33]: Creation complete after 1s [id=1feeeea8-697f-47d5-a0e8-3c0c810fb9f1]
pagerduty_ruleset_rule.a[48]: Creation complete after 1s [id=342cfe7e-21c8-402b-bed9-ea7b4a73ba91]
pagerduty_ruleset_rule.a[3]: Creating...
pagerduty_ruleset_rule.a[59]: Creating...
pagerduty_ruleset_rule.a[88]: Creation complete after 1s [id=ab5e05f0-28dd-473c-94eb-c1e14c5c3038]
pagerduty_ruleset_rule.a[45]: Creating...
pagerduty_ruleset_rule.a[38]: Creation complete after 4s [id=e772bd6a-cf58-42f3-afb7-60735175a112]
pagerduty_ruleset_rule.a[26]: Creating...
pagerduty_ruleset_rule.a[23]: Creation complete after 1s [id=da7d0b89-5587-4756-9786-e1c87a576171]
pagerduty_ruleset_rule.a[71]: Creating...
pagerduty_ruleset_rule.a[10]: Creation complete after 1s [id=bb347b1c-d3f4-472d-9f76-ec36c2cefc35]
pagerduty_ruleset_rule.a[20]: Creating...
pagerduty_ruleset_rule.a[26]: Creation complete after 1s [id=48ad90c2-908f-470a-813d-3175d520b810]
pagerduty_ruleset_rule.a[3]: Creation complete after 1s [id=21008c9d-fdf9-4738-be0a-10251959b4c9]
pagerduty_ruleset_rule.a[22]: Creating...
pagerduty_ruleset_rule.a[67]: Creating...
pagerduty_ruleset_rule.a[19]: Creation complete after 4s [id=d270b58a-4667-4c9f-b312-7c73fb43beb8]
pagerduty_ruleset_rule.a[2]: Creating...
pagerduty_ruleset_rule.a[71]: Creation complete after 1s [id=f230978b-48ac-4b4f-b5e1-9a88c97b8298]
pagerduty_ruleset_rule.a[54]: Creating...
pagerduty_ruleset_rule.a[22]: Creation complete after 1s [id=063a3ab2-1cf2-46d9-b479-9e4086f3ef7e]
pagerduty_ruleset_rule.a[45]: Creation complete after 2s [id=74f9802d-ecec-40d1-b015-4e8e65245a38]
pagerduty_ruleset_rule.a[35]: Creating...
pagerduty_ruleset_rule.a[70]: Creating...
pagerduty_ruleset_rule.a[2]: Creation complete after 0s [id=2338d7ae-ab1b-4006-832f-9635d61a1b7d]
pagerduty_ruleset_rule.a[5]: Creating...
pagerduty_ruleset_rule.a[20]: Creation complete after 2s [id=1ecd4b4b-3fe6-4eeb-a89f-9e78645dfcac]
pagerduty_ruleset_rule.a[15]: Creating...
pagerduty_ruleset_rule.a[54]: Creation complete after 1s [id=e24a6b62-6e4d-47ef-99af-c365e48d8edf]
pagerduty_ruleset_rule.a[74]: Creating...
pagerduty_ruleset_rule.a[70]: Creation complete after 1s [id=4fad3435-0c9f-4edd-83ee-1fdb1dcc1886]
pagerduty_ruleset_rule.a[36]: Creating...
pagerduty_ruleset_rule.a[27]: Creation complete after 3s [id=b51c8a10-569e-40c6-aacc-855dca87cac1]
pagerduty_ruleset_rule.a[9]: Creating...
pagerduty_ruleset_rule.a[15]: Creation complete after 1s [id=b18df6d2-6375-49e3-b62e-851635848da7]
pagerduty_ruleset_rule.a[59]: Creation complete after 4s [id=9e07a4a1-c73e-4bb9-950a-a5b20b704eca]
pagerduty_ruleset_rule.a[63]: Creating...
pagerduty_ruleset_rule.a[66]: Creating...
pagerduty_ruleset_rule.a[74]: Creation complete after 1s [id=30624c68-03ea-4c0f-af37-6405d373a692]
pagerduty_ruleset_rule.a[30]: Creating...
pagerduty_ruleset_rule.a[36]: Creation complete after 1s [id=31c0a087-27f9-4470-b877-99c4cb17101b]
pagerduty_ruleset_rule.a[99]: Creating...
pagerduty_ruleset_rule.a[9]: Creation complete after 1s [id=6a4782e8-d49c-43b1-8984-64ee5f2a6e1a]
pagerduty_ruleset_rule.a[80]: Creating...
pagerduty_ruleset_rule.a[63]: Creation complete after 1s [id=378f7e18-9a44-4b60-a25d-66b66d3891d4]
pagerduty_ruleset_rule.a[81]: Creating...
pagerduty_ruleset_rule.a[99]: Creation complete after 1s [id=2ebc3d73-7166-4277-9909-0a0d409c9336]
pagerduty_ruleset_rule.a[87]: Creating...
pagerduty_ruleset_rule.a[80]: Creation complete after 1s [id=aa80f026-93fa-4717-8381-12fcc08f2c19]
pagerduty_ruleset_rule.a[49]: Creating...
pagerduty_ruleset_rule.a[81]: Creation complete after 0s [id=caf5e1f2-c7e0-45b9-b2c1-6c8164e52adf]
pagerduty_ruleset_rule.a[25]: Creating...
pagerduty_ruleset_rule.a[66]: Creation complete after 1s [id=32fb1b45-2f7d-4f13-a851-c9086d65223f]
pagerduty_ruleset_rule.a[53]: Creating...
pagerduty_ruleset_rule.a[87]: Creation complete after 0s [id=18ca9e67-25c8-4537-a383-0c89a307a9cc]
pagerduty_ruleset_rule.a[8]: Creating...
pagerduty_ruleset_rule.a[49]: Creation complete after 1s [id=7bc9ff0e-1113-4ef6-9f26-b282ddc731cb]
pagerduty_ruleset_rule.a[73]: Creating...
pagerduty_ruleset_rule.a[73]: Creation complete after 0s [id=1fb27c92-0120-4cc0-97e8-c173b7017d23]
pagerduty_ruleset_rule.a[60]: Creating...
pagerduty_ruleset_rule.a[35]: Creation complete after 4s [id=194ba2a6-0332-4ac8-9832-bab0d4c834f2]
pagerduty_ruleset_rule.a[7]: Creating...
pagerduty_ruleset_rule.a[18]: Still creating... [10s elapsed]
pagerduty_ruleset_rule.a[82]: Still creating... [10s elapsed]
pagerduty_ruleset_rule.a[30]: Creation complete after 3s [id=e323d341-372b-439e-8a8d-f9f990cb21e8]
pagerduty_ruleset_rule.a[25]: Creation complete after 2s [id=216d3c19-d8bf-4b18-9a41-45621257ec97]
pagerduty_ruleset_rule.a[11]: Creating...
pagerduty_ruleset_rule.a[89]: Creating...
pagerduty_ruleset_rule.a[67]: Creation complete after 6s [id=45cf10d3-cc17-458b-8427-5b5bdaebe55e]
pagerduty_ruleset_rule.a[29]: Creating...
pagerduty_ruleset_rule.a[60]: Creation complete after 1s [id=b1f6a4ce-57c7-43db-8926-f189a3251c9b]
pagerduty_ruleset_rule.a[34]: Creating...
pagerduty_ruleset_rule.a[11]: Creation complete after 1s [id=c6fcfcf1-09ff-4bea-92ab-0aca3c5b0567]
pagerduty_ruleset_rule.a[17]: Creating...
pagerduty_ruleset_rule.a[8]: Creation complete after 3s [id=9b36cce7-1353-400d-8c05-5143451291bd]
pagerduty_ruleset_rule.a[72]: Creating...
pagerduty_ruleset_rule.a[34]: Creation complete after 1s [id=8b8f515e-33bb-476a-9963-c519dab748a3]
pagerduty_ruleset_rule.a[40]: Creating...
pagerduty_ruleset_rule.a[89]: Creation complete after 1s [id=c9472c0e-3314-47cd-9267-f1ddae1341b1]
pagerduty_ruleset_rule.a[78]: Creating...
pagerduty_ruleset_rule.a[82]: Creation complete after 12s [id=348c1b05-73fd-4512-b3c0-ea352c5d482c]
pagerduty_ruleset_rule.a[28]: Creating...
pagerduty_ruleset_rule.a[18]: Creation complete after 13s [id=0fe37087-200e-452a-90a9-18fadb4e8497]
pagerduty_ruleset_rule.a[72]: Creation complete after 1s [id=fa429d0e-c281-40a1-b08e-00df73bd3c9e]
pagerduty_ruleset_rule.a[40]: Creation complete after 1s [id=f8c6a8c1-6357-4a6e-8324-6c87189257aa]
pagerduty_ruleset_rule.a[42]: Creating...
pagerduty_ruleset_rule.a[79]: Creating...
pagerduty_ruleset_rule.a[91]: Creating...
pagerduty_ruleset_rule.a[79]: Creation complete after 1s [id=3cc32754-febb-49c7-b239-350194b55dc3]
pagerduty_ruleset_rule.a[97]: Creating...
pagerduty_ruleset_rule.a[29]: Creation complete after 3s [id=abfdbfb9-378f-405d-8464-afce41c6c993]
pagerduty_ruleset_rule.a[21]: Creating...
pagerduty_ruleset_rule.a[78]: Creation complete after 2s [id=61f421f6-67e3-4c1b-ae2c-7775dac289fb]
pagerduty_ruleset_rule.a[69]: Creating...
pagerduty_ruleset_rule.a[7]: Creation complete after 4s [id=1f18accf-bb5b-41c0-8da8-60024fd43cd0]
pagerduty_ruleset_rule.a[85]: Creating...
pagerduty_ruleset_rule.a[42]: Creation complete after 1s [id=deaff57d-5d86-49bf-a23a-7add7c708d7e]
pagerduty_ruleset_rule.a[65]: Creating...
pagerduty_ruleset_rule.a[53]: Creation complete after 5s [id=c01a86fa-51ee-4ad0-9d3a-fc0e703bd557]
pagerduty_ruleset_rule.a[43]: Creating...
pagerduty_ruleset_rule.a[21]: Creation complete after 2s [id=3c850163-f9dc-4508-9230-74c544cc1bdf]
pagerduty_ruleset_rule.a[12]: Creating...
pagerduty_ruleset_rule.a[69]: Creation complete after 2s [id=79d1992c-2835-4ac7-a84c-f13f4ae579ee]
pagerduty_ruleset_rule.a[56]: Creating...
pagerduty_ruleset_rule.a[28]: Creation complete after 4s [id=ba546c55-b3fe-406c-a801-20188ded0920]
pagerduty_ruleset_rule.a[90]: Creating...
pagerduty_ruleset_rule.a[65]: Creation complete after 2s [id=977fd73b-e1eb-4126-bab2-f18e8bae59dc]
pagerduty_ruleset_rule.a[31]: Creating...
pagerduty_ruleset_rule.a[5]: Creation complete after 10s [id=f2efdcd1-94a6-4c9d-92a8-3177cefe05ce]
pagerduty_ruleset_rule.a[84]: Creating...
pagerduty_ruleset_rule.a[56]: Creation complete after 0s [id=ad3738e1-f99f-467c-839d-e8c19fbee220]
pagerduty_ruleset_rule.a[13]: Creating...
pagerduty_ruleset_rule.a[85]: Creation complete after 3s [id=71d7323b-c8cd-477d-b3bd-9e94bc0c1c97]
pagerduty_ruleset_rule.a[64]: Creating...
pagerduty_ruleset_rule.a[31]: Creation complete after 1s [id=496f1c91-3d0c-4ef9-8920-069690e51327]
pagerduty_ruleset_rule.a[62]: Creating...
pagerduty_ruleset_rule.a[97]: Creation complete after 3s [id=e6417fce-95ec-4e87-b1c7-11a939314d6f]
pagerduty_ruleset_rule.a[58]: Creating...
pagerduty_ruleset_rule.a[12]: Creation complete after 1s [id=2373afe8-63ae-47e0-a2ed-48d30ae043d7]
pagerduty_ruleset_rule.a[92]: Creating...
pagerduty_ruleset_rule.a[90]: Creation complete after 2s [id=cfdbc945-0c6d-4d25-b66f-9c6e7f2d4e4c]
pagerduty_ruleset_rule.a[16]: Creating...
pagerduty_ruleset_rule.a[84]: Creation complete after 2s [id=b6530928-3155-4b80-b5e2-fec9fee5dfe5]
pagerduty_ruleset_rule.a[44]: Creating...
pagerduty_ruleset_rule.a[17]: Creation complete after 6s [id=b51f8df3-b652-4b1d-8ab6-ea811a7ed949]
pagerduty_ruleset_rule.a[96]: Creating...
pagerduty_ruleset_rule.a[16]: Creation complete after 1s [id=5cfef7d8-d188-4792-b00c-3a66e03cabd3]
pagerduty_ruleset_rule.a[86]: Creating...
pagerduty_ruleset_rule.a[92]: Creation complete after 2s [id=e1799688-fc92-43b3-b8fe-67053128b2e8]
pagerduty_ruleset_rule.a[94]: Creating...
pagerduty_ruleset_rule.a[58]: Creation complete after 2s [id=959e2627-12d1-4b92-8baa-1b31ef666417]
pagerduty_ruleset_rule.a[50]: Creating...
pagerduty_ruleset_rule.a[86]: Creation complete after 0s [id=d01918c2-565e-42d5-b8a7-ec44d7bb3580]
pagerduty_ruleset_rule.a[13]: Creation complete after 3s [id=3b2b3dd3-2a29-42fd-a5ae-10558de38d5c]
pagerduty_ruleset_rule.a[98]: Creating...
pagerduty_ruleset_rule.a[46]: Creating...
pagerduty_ruleset_rule.a[94]: Creation complete after 1s [id=34bf79cc-b230-494b-baeb-118bf0e93bf8]
pagerduty_ruleset_rule.a[32]: Creating...
pagerduty_ruleset_rule.a[62]: Creation complete after 3s [id=65d4b30c-5248-4ebd-a08b-f216bd6666ce]
pagerduty_ruleset_rule.a[4]: Creating...
pagerduty_ruleset_rule.a[50]: Creation complete after 1s [id=926fd2af-360a-4e51-bc13-08ab55da0ed5]
pagerduty_ruleset_rule.a[37]: Creating...
pagerduty_ruleset_rule.a[64]: Creation complete after 3s [id=9e7c6034-4482-4870-ac79-f04171189cb5]
pagerduty_ruleset_rule.a[95]: Creating...
pagerduty_ruleset_rule.a[32]: Creation complete after 0s [id=fdace580-750c-4a37-91c7-592339dee97c]
pagerduty_ruleset_rule.a[77]: Creating...
pagerduty_ruleset_rule.a[4]: Creation complete after 1s [id=11af20b5-f36f-4ef7-8b64-25221512e946]
pagerduty_ruleset_rule.a[41]: Creating...
pagerduty_ruleset_rule.a[43]: Creation complete after 7s [id=50527c67-0d5f-4e83-aa0a-e91e2bd7270d]
pagerduty_ruleset_rule.a[52]: Creating...
pagerduty_ruleset_rule.a[98]: Creation complete after 2s [id=13002641-66ae-4e69-ac32-a2f9d06d1e94]
pagerduty_ruleset_rule.a[77]: Creation complete after 1s [id=5be473ed-3a54-4bb8-85b2-7b192181ffb9]
pagerduty_ruleset_rule.a[41]: Creation complete after 0s [id=62f51a24-fab6-47ff-8f4c-3c9b9bf73cd8]
pagerduty_ruleset_rule.a[52]: Creation complete after 0s [id=1d251fb6-486d-41b5-9aae-f14c3d2c7b40]
pagerduty_ruleset_rule.a[96]: Creation complete after 4s [id=fb9e4637-f652-4f0f-9c9b-30c5fcb874de]
pagerduty_ruleset_rule.a[44]: Creation complete after 4s [id=d65a0341-8f0a-4007-bca5-c785d985b14b]
pagerduty_ruleset_rule.a[46]: Creation complete after 3s [id=4b8ed067-49d6-4944-ac3c-7240198ed8dc]
pagerduty_ruleset_rule.a[91]: Still creating... [10s elapsed]
pagerduty_ruleset_rule.a[95]: Creation complete after 3s [id=bc6aea98-89a3-43d0-a44f-2a1c93c08efe]
pagerduty_ruleset_rule.a[37]: Creation complete after 3s [id=03bb0a00-f674-42e5-a704-cb6e024ee4bc]
pagerduty_ruleset_rule.a[91]: Creation complete after 11s [id=837d8257-aab3-45e6-8e4a-fadf4799baed]

Apply complete! Resources: 98 added, 1 changed, 0 destroyed.
➜  test-tf git:(master) ✗ terraform apply -auto-approve
pagerduty_ruleset_rule.a[3]: Refreshing state... [id=21008c9d-fdf9-4738-be0a-10251959b4c9]
pagerduty_ruleset_rule.a[88]: Refreshing state... [id=ab5e05f0-28dd-473c-94eb-c1e14c5c3038]
pagerduty_ruleset_rule.a[17]: Refreshing state... [id=b51f8df3-b652-4b1d-8ab6-ea811a7ed949]
pagerduty_ruleset_rule.a[57]: Refreshing state... [id=fd6cead2-177e-4bfc-a818-ed66100bba6a]
pagerduty_ruleset_rule.a[72]: Refreshing state... [id=fa429d0e-c281-40a1-b08e-00df73bd3c9e]
pagerduty_ruleset_rule.a[8]: Refreshing state... [id=9b36cce7-1353-400d-8c05-5143451291bd]
pagerduty_ruleset_rule.a[43]: Refreshing state... [id=50527c67-0d5f-4e83-aa0a-e91e2bd7270d]
pagerduty_ruleset_rule.a[31]: Refreshing state... [id=496f1c91-3d0c-4ef9-8920-069690e51327]
pagerduty_ruleset_rule.a[77]: Refreshing state... [id=5be473ed-3a54-4bb8-85b2-7b192181ffb9]
pagerduty_ruleset_rule.a[0]: Refreshing state... [id=fd8d04ff-de0e-4986-bb1c-da7e0d94d9d7]
pagerduty_ruleset_rule.a[20]: Refreshing state... [id=1ecd4b4b-3fe6-4eeb-a89f-9e78645dfcac]
pagerduty_ruleset_rule.a[32]: Refreshing state... [id=fdace580-750c-4a37-91c7-592339dee97c]
pagerduty_ruleset_rule.a[56]: Refreshing state... [id=ad3738e1-f99f-467c-839d-e8c19fbee220]
pagerduty_ruleset_rule.a[41]: Refreshing state... [id=62f51a24-fab6-47ff-8f4c-3c9b9bf73cd8]
pagerduty_ruleset_rule.a[79]: Refreshing state... [id=3cc32754-febb-49c7-b239-350194b55dc3]
pagerduty_ruleset_rule.a[9]: Refreshing state... [id=6a4782e8-d49c-43b1-8984-64ee5f2a6e1a]
pagerduty_ruleset_rule.a[53]: Refreshing state... [id=c01a86fa-51ee-4ad0-9d3a-fc0e703bd557]
pagerduty_ruleset_rule.a[39]: Refreshing state... [id=29ec7e33-be89-4f36-8c56-e08b0f709435]
pagerduty_ruleset_rule.a[49]: Refreshing state... [id=7bc9ff0e-1113-4ef6-9f26-b282ddc731cb]
pagerduty_ruleset_rule.a[62]: Refreshing state... [id=65d4b30c-5248-4ebd-a08b-f216bd6666ce]
pagerduty_ruleset_rule.a[91]: Refreshing state... [id=837d8257-aab3-45e6-8e4a-fadf4799baed]
pagerduty_ruleset_rule.a[37]: Refreshing state... [id=03bb0a00-f674-42e5-a704-cb6e024ee4bc]
pagerduty_ruleset_rule.a[83]: Refreshing state... [id=9b909eef-531d-4b9f-a062-3339688cfc7e]
pagerduty_ruleset_rule.a[70]: Refreshing state... [id=4fad3435-0c9f-4edd-83ee-1fdb1dcc1886]
pagerduty_ruleset_rule.a[74]: Refreshing state... [id=30624c68-03ea-4c0f-af37-6405d373a692]
pagerduty_ruleset_rule.a[51]: Refreshing state... [id=7d28eed7-8c4b-4f1e-90ef-33298f70deb3]
pagerduty_ruleset_rule.a[95]: Refreshing state... [id=bc6aea98-89a3-43d0-a44f-2a1c93c08efe]
pagerduty_ruleset_rule.a[80]: Refreshing state... [id=aa80f026-93fa-4717-8381-12fcc08f2c19]
pagerduty_ruleset_rule.a[1]: Refreshing state... [id=245da58a-d4a7-44e8-986d-a1487f59ca6e]
pagerduty_ruleset_rule.a[65]: Refreshing state... [id=977fd73b-e1eb-4126-bab2-f18e8bae59dc]
pagerduty_ruleset_rule.a[35]: Refreshing state... [id=194ba2a6-0332-4ac8-9832-bab0d4c834f2]
pagerduty_ruleset_rule.a[21]: Refreshing state... [id=3c850163-f9dc-4508-9230-74c544cc1bdf]
pagerduty_ruleset_rule.a[82]: Refreshing state... [id=348c1b05-73fd-4512-b3c0-ea352c5d482c]
pagerduty_ruleset_rule.a[86]: Refreshing state... [id=d01918c2-565e-42d5-b8a7-ec44d7bb3580]
pagerduty_ruleset_rule.a[89]: Refreshing state... [id=c9472c0e-3314-47cd-9267-f1ddae1341b1]
pagerduty_ruleset_rule.a[29]: Refreshing state... [id=abfdbfb9-378f-405d-8464-afce41c6c993]
pagerduty_ruleset_rule.a[87]: Refreshing state... [id=18ca9e67-25c8-4537-a383-0c89a307a9cc]
pagerduty_ruleset_rule.a[15]: Refreshing state... [id=b18df6d2-6375-49e3-b62e-851635848da7]
pagerduty_ruleset_rule.a[2]: Refreshing state... [id=2338d7ae-ab1b-4006-832f-9635d61a1b7d]
pagerduty_ruleset_rule.a[58]: Refreshing state... [id=959e2627-12d1-4b92-8baa-1b31ef666417]
pagerduty_ruleset_rule.a[61]: Refreshing state... [id=060f410c-02e7-4137-8569-c94d04cd160e]
pagerduty_ruleset_rule.a[54]: Refreshing state... [id=e24a6b62-6e4d-47ef-99af-c365e48d8edf]
pagerduty_ruleset_rule.a[78]: Refreshing state... [id=61f421f6-67e3-4c1b-ae2c-7775dac289fb]
pagerduty_ruleset_rule.a[84]: Refreshing state... [id=b6530928-3155-4b80-b5e2-fec9fee5dfe5]
pagerduty_ruleset_rule.a[81]: Refreshing state... [id=caf5e1f2-c7e0-45b9-b2c1-6c8164e52adf]
pagerduty_ruleset_rule.a[52]: Refreshing state... [id=1d251fb6-486d-41b5-9aae-f14c3d2c7b40]
pagerduty_ruleset_rule.a[44]: Refreshing state... [id=d65a0341-8f0a-4007-bca5-c785d985b14b]
pagerduty_ruleset_rule.a[96]: Refreshing state... [id=fb9e4637-f652-4f0f-9c9b-30c5fcb874de]
pagerduty_ruleset_rule.a[69]: Refreshing state... [id=79d1992c-2835-4ac7-a84c-f13f4ae579ee]
pagerduty_ruleset_rule.a[27]: Refreshing state... [id=b51c8a10-569e-40c6-aacc-855dca87cac1]
pagerduty_ruleset_rule.a[64]: Refreshing state... [id=9e7c6034-4482-4870-ac79-f04171189cb5]
pagerduty_ruleset_rule.a[30]: Refreshing state... [id=e323d341-372b-439e-8a8d-f9f990cb21e8]
pagerduty_ruleset_rule.a[85]: Refreshing state... [id=71d7323b-c8cd-477d-b3bd-9e94bc0c1c97]
pagerduty_ruleset_rule.a[25]: Refreshing state... [id=216d3c19-d8bf-4b18-9a41-45621257ec97]
pagerduty_ruleset_rule.a[97]: Refreshing state... [id=e6417fce-95ec-4e87-b1c7-11a939314d6f]
pagerduty_ruleset_rule.a[68]: Refreshing state... [id=496baada-2df4-4bee-bc30-1c8508665048]
pagerduty_ruleset_rule.a[98]: Refreshing state... [id=13002641-66ae-4e69-ac32-a2f9d06d1e94]
pagerduty_ruleset_rule.a[75]: Refreshing state... [id=ab4eece5-52b4-4f7f-8f49-13bc8f8de907]
pagerduty_ruleset_rule.a[10]: Refreshing state... [id=bb347b1c-d3f4-472d-9f76-ec36c2cefc35]
pagerduty_ruleset_rule.a[46]: Refreshing state... [id=4b8ed067-49d6-4944-ac3c-7240198ed8dc]
pagerduty_ruleset_rule.a[59]: Refreshing state... [id=9e07a4a1-c73e-4bb9-950a-a5b20b704eca]
pagerduty_ruleset_rule.a[71]: Refreshing state... [id=f230978b-48ac-4b4f-b5e1-9a88c97b8298]
pagerduty_ruleset_rule.a[16]: Refreshing state... [id=5cfef7d8-d188-4792-b00c-3a66e03cabd3]
pagerduty_ruleset_rule.a[33]: Refreshing state... [id=1feeeea8-697f-47d5-a0e8-3c0c810fb9f1]
pagerduty_ruleset_rule.a[13]: Refreshing state... [id=3b2b3dd3-2a29-42fd-a5ae-10558de38d5c]
pagerduty_ruleset_rule.a[42]: Refreshing state... [id=deaff57d-5d86-49bf-a23a-7add7c708d7e]
pagerduty_ruleset_rule.a[47]: Refreshing state... [id=89207966-e67b-4188-8a32-fc5a1f3e5086]
pagerduty_ruleset_rule.a[18]: Refreshing state... [id=0fe37087-200e-452a-90a9-18fadb4e8497]
pagerduty_ruleset_rule.a[60]: Refreshing state... [id=b1f6a4ce-57c7-43db-8926-f189a3251c9b]
pagerduty_ruleset_rule.a[66]: Refreshing state... [id=32fb1b45-2f7d-4f13-a851-c9086d65223f]
pagerduty_ruleset_rule.a[90]: Refreshing state... [id=cfdbc945-0c6d-4d25-b66f-9c6e7f2d4e4c]
pagerduty_ruleset_rule.a[99]: Refreshing state... [id=2ebc3d73-7166-4277-9909-0a0d409c9336]
pagerduty_ruleset_rule.a[73]: Refreshing state... [id=1fb27c92-0120-4cc0-97e8-c173b7017d23]
pagerduty_ruleset_rule.a[93]: Refreshing state... [id=43a53844-4738-44a9-b614-9b71ca420b83]
pagerduty_ruleset_rule.a[19]: Refreshing state... [id=d270b58a-4667-4c9f-b312-7c73fb43beb8]
pagerduty_ruleset_rule.a[94]: Refreshing state... [id=34bf79cc-b230-494b-baeb-118bf0e93bf8]
pagerduty_ruleset_rule.a[6]: Refreshing state... [id=35d18ee4-57b2-4049-a4bd-709a17920b0c]
pagerduty_ruleset_rule.a[50]: Refreshing state... [id=926fd2af-360a-4e51-bc13-08ab55da0ed5]
pagerduty_ruleset_rule.a[7]: Refreshing state... [id=1f18accf-bb5b-41c0-8da8-60024fd43cd0]
pagerduty_ruleset_rule.a[76]: Refreshing state... [id=38fe3be6-fb7d-4847-ad18-847512cb3bae]
pagerduty_ruleset_rule.a[92]: Refreshing state... [id=e1799688-fc92-43b3-b8fe-67053128b2e8]
pagerduty_ruleset_rule.a[26]: Refreshing state... [id=48ad90c2-908f-470a-813d-3175d520b810]
pagerduty_ruleset_rule.a[12]: Refreshing state... [id=2373afe8-63ae-47e0-a2ed-48d30ae043d7]
pagerduty_ruleset_rule.a[36]: Refreshing state... [id=31c0a087-27f9-4470-b877-99c4cb17101b]
pagerduty_ruleset_rule.a[55]: Refreshing state... [id=fcc1ccac-d5d1-4f72-8547-9c8be891252d]
pagerduty_ruleset_rule.a[23]: Refreshing state... [id=da7d0b89-5587-4756-9786-e1c87a576171]
pagerduty_ruleset_rule.a[67]: Refreshing state... [id=45cf10d3-cc17-458b-8427-5b5bdaebe55e]
pagerduty_ruleset_rule.a[34]: Refreshing state... [id=8b8f515e-33bb-476a-9963-c519dab748a3]
pagerduty_ruleset_rule.a[5]: Refreshing state... [id=f2efdcd1-94a6-4c9d-92a8-3177cefe05ce]
pagerduty_ruleset_rule.a[38]: Refreshing state... [id=e772bd6a-cf58-42f3-afb7-60735175a112]
pagerduty_ruleset_rule.a[4]: Refreshing state... [id=11af20b5-f36f-4ef7-8b64-25221512e946]
pagerduty_ruleset_rule.a[14]: Refreshing state... [id=40f2d89f-8eb3-49d4-a546-81a4f6f4d186]
pagerduty_ruleset_rule.a[63]: Refreshing state... [id=378f7e18-9a44-4b60-a25d-66b66d3891d4]
pagerduty_ruleset_rule.a[48]: Refreshing state... [id=342cfe7e-21c8-402b-bed9-ea7b4a73ba91]
pagerduty_ruleset_rule.a[11]: Refreshing state... [id=c6fcfcf1-09ff-4bea-92ab-0aca3c5b0567]
pagerduty_ruleset_rule.a[40]: Refreshing state... [id=f8c6a8c1-6357-4a6e-8324-6c87189257aa]
pagerduty_ruleset_rule.a[28]: Refreshing state... [id=ba546c55-b3fe-406c-a801-20188ded0920]
pagerduty_ruleset_rule.a[24]: Refreshing state... [id=455c59dd-d82a-494f-bf40-247e64e54e06]
pagerduty_ruleset_rule.a[22]: Refreshing state... [id=063a3ab2-1cf2-46d9-b479-9e4086f3ef7e]
pagerduty_ruleset_rule.a[45]: Refreshing state... [id=74f9802d-ecec-40d1-b015-4e8e65245a38]
pagerduty_ruleset_rule.a[26]: Destroying... [id=48ad90c2-908f-470a-813d-3175d520b810]
pagerduty_ruleset_rule.a[78]: Destroying... [id=61f421f6-67e3-4c1b-ae2c-7775dac289fb]
pagerduty_ruleset_rule.a[83]: Destroying... [id=9b909eef-531d-4b9f-a062-3339688cfc7e]
pagerduty_ruleset_rule.a[27]: Destroying... [id=b51c8a10-569e-40c6-aacc-855dca87cac1]
pagerduty_ruleset_rule.a[98]: Destroying... [id=13002641-66ae-4e69-ac32-a2f9d06d1e94]
pagerduty_ruleset_rule.a[46]: Destroying... [id=4b8ed067-49d6-4944-ac3c-7240198ed8dc]
pagerduty_ruleset_rule.a[54]: Destroying... [id=e24a6b62-6e4d-47ef-99af-c365e48d8edf]
pagerduty_ruleset_rule.a[82]: Destroying... [id=348c1b05-73fd-4512-b3c0-ea352c5d482c]
pagerduty_ruleset_rule.a[43]: Destroying... [id=50527c67-0d5f-4e83-aa0a-e91e2bd7270d]
pagerduty_ruleset_rule.a[87]: Destroying... [id=18ca9e67-25c8-4537-a383-0c89a307a9cc]
pagerduty_ruleset_rule.a[78]: Destruction complete after 1s
pagerduty_ruleset_rule.a[63]: Destroying... [id=378f7e18-9a44-4b60-a25d-66b66d3891d4]
pagerduty_ruleset_rule.a[26]: Destruction complete after 1s
pagerduty_ruleset_rule.a[91]: Destroying... [id=837d8257-aab3-45e6-8e4a-fadf4799baed]
pagerduty_ruleset_rule.a[27]: Destruction complete after 1s
pagerduty_ruleset_rule.a[65]: Destroying... [id=977fd73b-e1eb-4126-bab2-f18e8bae59dc]
pagerduty_ruleset_rule.a[83]: Destruction complete after 1s
pagerduty_ruleset_rule.a[17]: Destroying... [id=b51f8df3-b652-4b1d-8ab6-ea811a7ed949]
pagerduty_ruleset_rule.a[65]: Destruction complete after 1s
pagerduty_ruleset_rule.a[93]: Destroying... [id=43a53844-4738-44a9-b614-9b71ca420b83]
pagerduty_ruleset_rule.a[17]: Destruction complete after 1s
pagerduty_ruleset_rule.a[90]: Destroying... [id=cfdbc945-0c6d-4d25-b66f-9c6e7f2d4e4c]
pagerduty_ruleset_rule.a[82]: Destruction complete after 2s
pagerduty_ruleset_rule.a[73]: Destroying... [id=1fb27c92-0120-4cc0-97e8-c173b7017d23]
pagerduty_ruleset_rule.a[87]: Destruction complete after 2s
pagerduty_ruleset_rule.a[99]: Destroying... [id=2ebc3d73-7166-4277-9909-0a0d409c9336]
pagerduty_ruleset_rule.a[73]: Destruction complete after 0s
pagerduty_ruleset_rule.a[79]: Destroying... [id=3cc32754-febb-49c7-b239-350194b55dc3]
pagerduty_ruleset_rule.a[99]: Destruction complete after 1s
pagerduty_ruleset_rule.a[49]: Destroying... [id=7bc9ff0e-1113-4ef6-9f26-b282ddc731cb]
pagerduty_ruleset_rule.a[90]: Destruction complete after 1s
pagerduty_ruleset_rule.a[89]: Destroying... [id=c9472c0e-3314-47cd-9267-f1ddae1341b1]
pagerduty_ruleset_rule.a[49]: Destruction complete after 0s
pagerduty_ruleset_rule.a[41]: Destroying... [id=62f51a24-fab6-47ff-8f4c-3c9b9bf73cd8]
pagerduty_ruleset_rule.a[89]: Destruction complete after 0s
pagerduty_ruleset_rule.a[97]: Destroying... [id=e6417fce-95ec-4e87-b1c7-11a939314d6f]
pagerduty_ruleset_rule.a[63]: Destruction complete after 3s
pagerduty_ruleset_rule.a[25]: Destroying... [id=216d3c19-d8bf-4b18-9a41-45621257ec97]
pagerduty_ruleset_rule.a[98]: Destruction complete after 4s
pagerduty_ruleset_rule.a[22]: Destroying... [id=063a3ab2-1cf2-46d9-b479-9e4086f3ef7e]
pagerduty_ruleset_rule.a[25]: Destruction complete after 0s
pagerduty_ruleset_rule.a[77]: Destroying... [id=5be473ed-3a54-4bb8-85b2-7b192181ffb9]
pagerduty_ruleset_rule.a[22]: Destruction complete after 0s
pagerduty_ruleset_rule.a[4]: Destroying... [id=11af20b5-f36f-4ef7-8b64-25221512e946]
pagerduty_ruleset_rule.a[93]: Destruction complete after 2s
pagerduty_ruleset_rule.a[13]: Destroying... [id=3b2b3dd3-2a29-42fd-a5ae-10558de38d5c]
pagerduty_ruleset_rule.a[97]: Destruction complete after 2s
pagerduty_ruleset_rule.a[95]: Destroying... [id=bc6aea98-89a3-43d0-a44f-2a1c93c08efe]
pagerduty_ruleset_rule.a[13]: Destruction complete after 1s
pagerduty_ruleset_rule.a[31]: Destroying... [id=496f1c91-3d0c-4ef9-8920-069690e51327]
pagerduty_ruleset_rule.a[77]: Destruction complete after 1s
pagerduty_ruleset_rule.a[59]: Destroying... [id=9e07a4a1-c73e-4bb9-950a-a5b20b704eca]
pagerduty_ruleset_rule.a[4]: Destruction complete after 1s
pagerduty_ruleset_rule.a[10]: Destroying... [id=bb347b1c-d3f4-472d-9f76-ec36c2cefc35]
pagerduty_ruleset_rule.a[46]: Destruction complete after 5s
pagerduty_ruleset_rule.a[29]: Destroying... [id=abfdbfb9-378f-405d-8464-afce41c6c993]
pagerduty_ruleset_rule.a[43]: Destruction complete after 6s
pagerduty_ruleset_rule.a[92]: Destroying... [id=e1799688-fc92-43b3-b8fe-67053128b2e8]
pagerduty_ruleset_rule.a[41]: Destruction complete after 3s
pagerduty_ruleset_rule.a[64]: Destroying... [id=9e7c6034-4482-4870-ac79-f04171189cb5]
pagerduty_ruleset_rule.a[10]: Destruction complete after 1s
pagerduty_ruleset_rule.a[67]: Destroying... [id=45cf10d3-cc17-458b-8427-5b5bdaebe55e]
pagerduty_ruleset_rule.a[31]: Destruction complete after 1s
pagerduty_ruleset_rule.a[14]: Destroying... [id=40f2d89f-8eb3-49d4-a546-81a4f6f4d186]
pagerduty_ruleset_rule.a[92]: Destruction complete after 0s
pagerduty_ruleset_rule.a[71]: Destroying... [id=f230978b-48ac-4b4f-b5e1-9a88c97b8298]
pagerduty_ruleset_rule.a[14]: Destruction complete after 0s
pagerduty_ruleset_rule.a[23]: Destroying... [id=da7d0b89-5587-4756-9786-e1c87a576171]
pagerduty_ruleset_rule.a[64]: Destruction complete after 0s
pagerduty_ruleset_rule.a[19]: Destroying... [id=d270b58a-4667-4c9f-b312-7c73fb43beb8]
pagerduty_ruleset_rule.a[23]: Destruction complete after 1s
pagerduty_ruleset_rule.a[16]: Destroying... [id=5cfef7d8-d188-4792-b00c-3a66e03cabd3]
pagerduty_ruleset_rule.a[19]: Destruction complete after 0s
pagerduty_ruleset_rule.a[18]: Destroying... [id=0fe37087-200e-452a-90a9-18fadb4e8497]
pagerduty_ruleset_rule.a[95]: Destruction complete after 2s
pagerduty_ruleset_rule.a[72]: Destroying... [id=fa429d0e-c281-40a1-b08e-00df73bd3c9e]
pagerduty_ruleset_rule.a[18]: Destruction complete after 0s
pagerduty_ruleset_rule.a[3]: Destroying... [id=21008c9d-fdf9-4738-be0a-10251959b4c9]
pagerduty_ruleset_rule.a[72]: Destruction complete after 1s
pagerduty_ruleset_rule.a[48]: Destroying... [id=342cfe7e-21c8-402b-bed9-ea7b4a73ba91]
pagerduty_ruleset_rule.a[71]: Destruction complete after 2s
pagerduty_ruleset_rule.a[94]: Destroying... [id=34bf79cc-b230-494b-baeb-118bf0e93bf8]
pagerduty_ruleset_rule.a[16]: Destruction complete after 1s
pagerduty_ruleset_rule.a[58]: Destroying... [id=959e2627-12d1-4b92-8baa-1b31ef666417]
pagerduty_ruleset_rule.a[94]: Destruction complete after 0s
pagerduty_ruleset_rule.a[47]: Destroying... [id=89207966-e67b-4188-8a32-fc5a1f3e5086]
pagerduty_ruleset_rule.a[3]: Destruction complete after 1s
pagerduty_ruleset_rule.a[7]: Destroying... [id=1f18accf-bb5b-41c0-8da8-60024fd43cd0]
pagerduty_ruleset_rule.a[67]: Destruction complete after 3s
pagerduty_ruleset_rule.a[35]: Destroying... [id=194ba2a6-0332-4ac8-9832-bab0d4c834f2]
pagerduty_ruleset_rule.a[7]: Destruction complete after 1s
pagerduty_ruleset_rule.a[57]: Destroying... [id=fd6cead2-177e-4bfc-a818-ed66100bba6a]
pagerduty_ruleset_rule.a[35]: Destruction complete after 0s
pagerduty_ruleset_rule.a[2]: Destroying... [id=2338d7ae-ab1b-4006-832f-9635d61a1b7d]
pagerduty_ruleset_rule.a[57]: Destruction complete after 0s
pagerduty_ruleset_rule.a[69]: Destroying... [id=79d1992c-2835-4ac7-a84c-f13f4ae579ee]
pagerduty_ruleset_rule.a[2]: Destruction complete after 0s
pagerduty_ruleset_rule.a[11]: Destroying... [id=c6fcfcf1-09ff-4bea-92ab-0aca3c5b0567]
pagerduty_ruleset_rule.a[69]: Destruction complete after 0s
pagerduty_ruleset_rule.a[52]: Destroying... [id=1d251fb6-486d-41b5-9aae-f14c3d2c7b40]
pagerduty_ruleset_rule.a[11]: Destruction complete after 0s
pagerduty_ruleset_rule.a[86]: Destroying... [id=d01918c2-565e-42d5-b8a7-ec44d7bb3580]
pagerduty_ruleset_rule.a[52]: Destruction complete after 1s
pagerduty_ruleset_rule.a[40]: Destroying... [id=f8c6a8c1-6357-4a6e-8324-6c87189257aa]
pagerduty_ruleset_rule.a[86]: Destruction complete after 0s
pagerduty_ruleset_rule.a[45]: Destroying... [id=74f9802d-ecec-40d1-b015-4e8e65245a38]
pagerduty_ruleset_rule.a[48]: Destruction complete after 2s
pagerduty_ruleset_rule.a[50]: Destroying... [id=926fd2af-360a-4e51-bc13-08ab55da0ed5]
pagerduty_ruleset_rule.a[59]: Destruction complete after 5s
pagerduty_ruleset_rule.a[34]: Destroying... [id=8b8f515e-33bb-476a-9963-c519dab748a3]
pagerduty_ruleset_rule.a[45]: Destruction complete after 0s
pagerduty_ruleset_rule.a[32]: Destroying... [id=fdace580-750c-4a37-91c7-592339dee97c]
pagerduty_ruleset_rule.a[91]: Destruction complete after 9s
pagerduty_ruleset_rule.a[44]: Destroying... [id=d65a0341-8f0a-4007-bca5-c785d985b14b]
pagerduty_ruleset_rule.a[54]: Still destroying... [id=e24a6b62-6e4d-47ef-99af-c365e48d8edf, 10s elapsed]
pagerduty_ruleset_rule.a[29]: Destruction complete after 5s
pagerduty_ruleset_rule.a[55]: Destroying... [id=fcc1ccac-d5d1-4f72-8547-9c8be891252d]
pagerduty_ruleset_rule.a[54]: Destruction complete after 11s
pagerduty_ruleset_rule.a[68]: Destroying... [id=496baada-2df4-4bee-bc30-1c8508665048]
pagerduty_ruleset_rule.a[44]: Destruction complete after 1s
pagerduty_ruleset_rule.a[56]: Destroying... [id=ad3738e1-f99f-467c-839d-e8c19fbee220]
pagerduty_ruleset_rule.a[68]: Destruction complete after 0s
pagerduty_ruleset_rule.a[21]: Destroying... [id=3c850163-f9dc-4508-9230-74c544cc1bdf]
pagerduty_ruleset_rule.a[79]: Destruction complete after 9s
pagerduty_ruleset_rule.a[1]: Destroying... [id=245da58a-d4a7-44e8-986d-a1487f59ca6e]
pagerduty_ruleset_rule.a[34]: Destruction complete after 1s
pagerduty_ruleset_rule.a[6]: Destroying... [id=35d18ee4-57b2-4049-a4bd-709a17920b0c]
pagerduty_ruleset_rule.a[21]: Destruction complete after 1s
pagerduty_ruleset_rule.a[0]: Destroying... [id=fd8d04ff-de0e-4986-bb1c-da7e0d94d9d7]
pagerduty_ruleset_rule.a[6]: Destruction complete after 1s
pagerduty_ruleset_rule.a[39]: Destroying... [id=29ec7e33-be89-4f36-8c56-e08b0f709435]
pagerduty_ruleset_rule.a[0]: Destruction complete after 0s
pagerduty_ruleset_rule.a[12]: Destroying... [id=2373afe8-63ae-47e0-a2ed-48d30ae043d7]
pagerduty_ruleset_rule.a[56]: Destruction complete after 1s
pagerduty_ruleset_rule.a[53]: Destroying... [id=c01a86fa-51ee-4ad0-9d3a-fc0e703bd557]
pagerduty_ruleset_rule.a[55]: Destruction complete after 1s
pagerduty_ruleset_rule.a[9]: Destroying... [id=6a4782e8-d49c-43b1-8984-64ee5f2a6e1a]
pagerduty_ruleset_rule.a[40]: Destruction complete after 2s
pagerduty_ruleset_rule.a[30]: Destroying... [id=e323d341-372b-439e-8a8d-f9f990cb21e8]
pagerduty_ruleset_rule.a[53]: Destruction complete after 1s
pagerduty_ruleset_rule.a[5]: Destroying... [id=f2efdcd1-94a6-4c9d-92a8-3177cefe05ce]
pagerduty_ruleset_rule.a[30]: Destruction complete after 1s
pagerduty_ruleset_rule.a[20]: Destroying... [id=1ecd4b4b-3fe6-4eeb-a89f-9e78645dfcac]
pagerduty_ruleset_rule.a[32]: Destruction complete after 3s
pagerduty_ruleset_rule.a[28]: Destroying... [id=ba546c55-b3fe-406c-a801-20188ded0920]
pagerduty_ruleset_rule.a[5]: Destruction complete after 0s
pagerduty_ruleset_rule.a[66]: Destroying... [id=32fb1b45-2f7d-4f13-a851-c9086d65223f]
pagerduty_ruleset_rule.a[47]: Destruction complete after 5s
pagerduty_ruleset_rule.a[74]: Destroying... [id=30624c68-03ea-4c0f-af37-6405d373a692]
pagerduty_ruleset_rule.a[12]: Destruction complete after 1s
pagerduty_ruleset_rule.a[60]: Destroying... [id=b1f6a4ce-57c7-43db-8926-f189a3251c9b]
pagerduty_ruleset_rule.a[9]: Destruction complete after 1s
pagerduty_ruleset_rule.a[96]: Destroying... [id=fb9e4637-f652-4f0f-9c9b-30c5fcb874de]
pagerduty_ruleset_rule.a[1]: Destruction complete after 3s
pagerduty_ruleset_rule.a[36]: Destroying... [id=31c0a087-27f9-4470-b877-99c4cb17101b]
pagerduty_ruleset_rule.a[96]: Destruction complete after 0s
pagerduty_ruleset_rule.a[62]: Destroying... [id=65d4b30c-5248-4ebd-a08b-f216bd6666ce]
pagerduty_ruleset_rule.a[36]: Destruction complete after 0s
pagerduty_ruleset_rule.a[51]: Destroying... [id=7d28eed7-8c4b-4f1e-90ef-33298f70deb3]
pagerduty_ruleset_rule.a[20]: Destruction complete after 1s
pagerduty_ruleset_rule.a[84]: Destroying... [id=b6530928-3155-4b80-b5e2-fec9fee5dfe5]
pagerduty_ruleset_rule.a[62]: Destruction complete after 0s
pagerduty_ruleset_rule.a[37]: Destroying... [id=03bb0a00-f674-42e5-a704-cb6e024ee4bc]
pagerduty_ruleset_rule.a[39]: Destruction complete after 2s
pagerduty_ruleset_rule.a[81]: Destroying... [id=caf5e1f2-c7e0-45b9-b2c1-6c8164e52adf]
pagerduty_ruleset_rule.a[37]: Destruction complete after 1s
pagerduty_ruleset_rule.a[88]: Destroying... [id=ab5e05f0-28dd-473c-94eb-c1e14c5c3038]
pagerduty_ruleset_rule.a[84]: Destruction complete after 1s
pagerduty_ruleset_rule.a[42]: Destroying... [id=deaff57d-5d86-49bf-a23a-7add7c708d7e]
pagerduty_ruleset_rule.a[50]: Destruction complete after 5s
pagerduty_ruleset_rule.a[33]: Destroying... [id=1feeeea8-697f-47d5-a0e8-3c0c810fb9f1]
pagerduty_ruleset_rule.a[88]: Destruction complete after 0s
pagerduty_ruleset_rule.a[38]: Destroying... [id=e772bd6a-cf58-42f3-afb7-60735175a112]
pagerduty_ruleset_rule.a[42]: Destruction complete after 0s
pagerduty_ruleset_rule.a[76]: Destroying... [id=38fe3be6-fb7d-4847-ad18-847512cb3bae]
pagerduty_ruleset_rule.a[28]: Destruction complete after 2s
pagerduty_ruleset_rule.a[80]: Destroying... [id=aa80f026-93fa-4717-8381-12fcc08f2c19]
pagerduty_ruleset_rule.a[38]: Destruction complete after 1s
pagerduty_ruleset_rule.a[85]: Destroying... [id=71d7323b-c8cd-477d-b3bd-9e94bc0c1c97]
pagerduty_ruleset_rule.a[76]: Destruction complete after 1s
pagerduty_ruleset_rule.a[75]: Destroying... [id=ab4eece5-52b4-4f7f-8f49-13bc8f8de907]
pagerduty_ruleset_rule.a[74]: Destruction complete after 3s
pagerduty_ruleset_rule.a[70]: Destroying... [id=4fad3435-0c9f-4edd-83ee-1fdb1dcc1886]
pagerduty_ruleset_rule.a[81]: Destruction complete after 2s
pagerduty_ruleset_rule.a[24]: Destroying... [id=455c59dd-d82a-494f-bf40-247e64e54e06]
pagerduty_ruleset_rule.a[33]: Destruction complete after 1s
pagerduty_ruleset_rule.a[61]: Destroying... [id=060f410c-02e7-4137-8569-c94d04cd160e]
pagerduty_ruleset_rule.a[75]: Destruction complete after 0s
pagerduty_ruleset_rule.a[8]: Destroying... [id=9b36cce7-1353-400d-8c05-5143451291bd]
pagerduty_ruleset_rule.a[70]: Destruction complete after 0s
pagerduty_ruleset_rule.a[15]: Destroying... [id=b18df6d2-6375-49e3-b62e-851635848da7]
pagerduty_ruleset_rule.a[85]: Destruction complete after 1s
pagerduty_ruleset_rule.a[51]: Destruction complete after 3s
pagerduty_ruleset_rule.a[8]: Destruction complete after 1s
pagerduty_ruleset_rule.a[61]: Destruction complete after 1s
pagerduty_ruleset_rule.a[24]: Destruction complete after 1s
pagerduty_ruleset_rule.a[15]: Destruction complete after 2s
pagerduty_ruleset_rule.a[66]: Destruction complete after 5s
pagerduty_ruleset_rule.a[58]: Still destroying... [id=959e2627-12d1-4b92-8baa-1b31ef666417, 10s elapsed]
pagerduty_ruleset_rule.a[80]: Destruction complete after 3s
pagerduty_ruleset_rule.a[60]: Destruction complete after 9s
pagerduty_ruleset_rule.a[58]: Destruction complete after 17s

Apply complete! Resources: 0 added, 0 changed, 100 destroyed.
```

**Comment by stmcallister** [comment]
Thank you both for reporting back! 

@cjdaniel -- yeah, i haven't touched the `pagerduty_event_rule` resource and don't plan to. The engineering teams at PagerDuty are going forward with adding features to the Ruleset Rules and will eventually deprecate the Event Rules endpoint at some point. 

Going to close this issue for now. Please feel free to repost or create a new Issue if you have any further issues.
